From e0d9deaab662468e11b08105e1b155660076b5eb Mon Sep 17 00:00:00 2001
From: Simon Pichugin <spichugi@redhat.com>
Date: Fri, 1 Aug 2025 09:28:39 -0700
Subject: [PATCH] Issues 6913, 6886, 6250 - Adjust xfail marks (#6914)

Description: Some of the ACI invalid syntax issues were fixed,
so we need to remove xfail marks.
Disk space issue should have a 'skipif' mark.
Display all attrs (nsslapd-auditlog-display-attrs: *) fails because of a bug.
EntryUSN inconsistency and overflow bugs were exposed with the tests.

Related: https://github.com/389ds/389-ds-base/issues/6913
Related: https://github.com/389ds/389-ds-base/issues/6886
Related: https://github.com/389ds/389-ds-base/issues/6250

Reviewed by: @vashirov (Thanks!)
---
 dirsrvtests/tests/suites/acl/syntax_test.py   | 13 ++++++++--
 .../tests/suites/import/regression_test.py    | 18 +++++++-------
 .../logging/audit_password_masking_test.py    | 24 +++++++++----------
 .../suites/plugins/entryusn_overflow_test.py  |  2 ++
 4 files changed, 34 insertions(+), 23 deletions(-)

diff --git a/dirsrvtests/tests/suites/acl/syntax_test.py b/dirsrvtests/tests/suites/acl/syntax_test.py
index 4edc7fa4b..ed9919ba3 100644
--- a/dirsrvtests/tests/suites/acl/syntax_test.py
+++ b/dirsrvtests/tests/suites/acl/syntax_test.py
@@ -190,10 +190,9 @@ FAILED = [('test_targattrfilters_18',
            f'(all)userdn="ldap:///anyone";)'), ]
 
 
-@pytest.mark.xfail(reason='https://bugzilla.redhat.com/show_bug.cgi?id=1691473')
 @pytest.mark.parametrize("real_value", [a[1] for a in FAILED],
                          ids=[a[0] for a in FAILED])
-def test_aci_invalid_syntax_fail(topo, real_value):
+def test_aci_invalid_syntax_fail(topo, real_value, request):
     """Try to set wrong ACI syntax.
 
         :id: 83c40784-fff5-49c8-9535-7064c9c19e7e
@@ -206,6 +205,16 @@ def test_aci_invalid_syntax_fail(topo, real_value):
             1. It should pass
             2. It should not pass
         """
+    # Mark specific test cases as xfail
+    xfail_cases = [
+        'test_targattrfilters_18',
+        'test_targattrfilters_20',
+        'test_bind_rule_set_with_more_than_three'
+    ]
+
+    if request.node.callspec.id in xfail_cases:
+        pytest.xfail("DS6913 - This test case is expected to fail")
+
     domain = Domain(topo.standalone, DEFAULT_SUFFIX)
     with pytest.raises(ldap.INVALID_SYNTAX):
         domain.add("aci", real_value)
diff --git a/dirsrvtests/tests/suites/import/regression_test.py b/dirsrvtests/tests/suites/import/regression_test.py
index e6fef89cc..61fdf8559 100644
--- a/dirsrvtests/tests/suites/import/regression_test.py
+++ b/dirsrvtests/tests/suites/import/regression_test.py
@@ -320,7 +320,7 @@ ou: myDups00001
     assert standalone.ds_error_log.match('.*Duplicated DN detected.*')
 
 @pytest.mark.tier2
-@pytest.mark.xfail(not _check_disk_space(), reason="not enough disk space for lmdb map")
+@pytest.mark.skipif(not _check_disk_space(), reason="not enough disk space for lmdb map")
 @pytest.mark.xfail(ds_is_older("1.3.10.1"), reason="bz1749595 not fixed on versions older than 1.3.10.1")
 def test_large_ldif2db_ancestorid_index_creation(topo, _set_mdb_map_size):
     """Import with ldif2db a large file - check that the ancestorid index creation phase has a correct performance
@@ -396,39 +396,39 @@ def test_large_ldif2db_ancestorid_index_creation(topo, _set_mdb_map_size):
     log.info('Starting the server')
     topo.standalone.start()
 
-    # With lmdb there is no more any special phase for ancestorid 
+    # With lmdb there is no more any special phase for ancestorid
     # because ancestorsid get updated on the fly while processing the
     # entryrdn (by up the parents chain to compute the parentid
-    # 
+    #
     # But there is still a numSubordinates generation phase
     if get_default_db_lib() == "mdb":
         log.info('parse the errors logs to check lines with "Generating numSubordinates complete." are present')
         end_numsubordinates = str(topo.standalone.ds_error_log.match(r'.*Generating numSubordinates complete.*'))[1:-1]
         assert len(end_numsubordinates) > 0
-    
+
     else:
         log.info('parse the errors logs to check lines with "Starting sort of ancestorid" are present')
         start_sort_str = str(topo.standalone.ds_error_log.match(r'.*Starting sort of ancestorid non-leaf IDs*'))[1:-1]
         assert len(start_sort_str) > 0
-    
+
         log.info('parse the errors logs to check lines with "Finished sort of ancestorid" are present')
         end_sort_str = str(topo.standalone.ds_error_log.match(r'.*Finished sort of ancestorid non-leaf IDs*'))[1:-1]
         assert len(end_sort_str) > 0
-    
+
         log.info('parse the error logs for the line with "Gathering ancestorid non-leaf IDs"')
         start_ancestorid_indexing_op_str = str(topo.standalone.ds_error_log.match(r'.*Gathering ancestorid non-leaf IDs*'))[1:-1]
         assert len(start_ancestorid_indexing_op_str) > 0
-    
+
         log.info('parse the error logs for the line with "Created ancestorid index"')
         end_ancestorid_indexing_op_str = str(topo.standalone.ds_error_log.match(r'.*Created ancestorid index*'))[1:-1]
         assert len(end_ancestorid_indexing_op_str) > 0
-    
+
         log.info('get the ancestorid non-leaf IDs indexing start and end time from the collected strings')
         # Collected lines look like : '[15/May/2020:05:30:27.245967313 -0400] - INFO - bdb_get_nonleaf_ids - import userRoot: Gathering ancestorid non-leaf IDs...'
         # We are getting the sec.nanosec part of the date, '27.245967313' in the above example
         start_time = (start_ancestorid_indexing_op_str.split()[0]).split(':')[3]
         end_time = (end_ancestorid_indexing_op_str.split()[0]).split(':')[3]
-    
+
         log.info('Calculate the elapsed time for the ancestorid non-leaf IDs index creation')
         etime = (Decimal(end_time) - Decimal(start_time))
         # The time for the ancestorid index creation should be less than 10s for an offline import of an ldif file with 100000 entries / 5 entries per node
diff --git a/dirsrvtests/tests/suites/logging/audit_password_masking_test.py b/dirsrvtests/tests/suites/logging/audit_password_masking_test.py
index 3b6a54849..69a36cb5d 100644
--- a/dirsrvtests/tests/suites/logging/audit_password_masking_test.py
+++ b/dirsrvtests/tests/suites/logging/audit_password_masking_test.py
@@ -117,10 +117,10 @@ def check_password_masked(inst, log_format, expected_password, actual_password):
 
 @pytest.mark.parametrize("log_format,display_attrs", [
     ("default", None),
-    ("default", "*"),
+    pytest.param("default", "*", marks=pytest.mark.xfail(reason="DS6886")),
     ("default", "userPassword"),
     ("json", None),
-    ("json", "*"),
+    pytest.param("json", "*", marks=pytest.mark.xfail(reason="DS6886")),
     ("json", "userPassword")
 ])
 def test_password_masking_add_operation(topo, log_format, display_attrs):
@@ -173,10 +173,10 @@ def test_password_masking_add_operation(topo, log_format, display_attrs):
 
 @pytest.mark.parametrize("log_format,display_attrs", [
     ("default", None),
-    ("default", "*"),
+    pytest.param("default", "*", marks=pytest.mark.xfail(reason="DS6886")),
     ("default", "userPassword"),
     ("json", None),
-    ("json", "*"),
+    pytest.param("json", "*", marks=pytest.mark.xfail(reason="DS6886")),
     ("json", "userPassword")
 ])
 def test_password_masking_modify_operation(topo, log_format, display_attrs):
@@ -242,10 +242,10 @@ def test_password_masking_modify_operation(topo, log_format, display_attrs):
 
 @pytest.mark.parametrize("log_format,display_attrs", [
     ("default", None),
-    ("default", "*"),
+    pytest.param("default", "*", marks=pytest.mark.xfail(reason="DS6886")),
     ("default", "nsslapd-rootpw"),
     ("json", None),
-    ("json", "*"),
+    pytest.param("json", "*", marks=pytest.mark.xfail(reason="DS6886")),
     ("json", "nsslapd-rootpw")
 ])
 def test_password_masking_rootpw_modify_operation(topo, log_format, display_attrs):
@@ -297,10 +297,10 @@ def test_password_masking_rootpw_modify_operation(topo, log_format, display_attr
 
 @pytest.mark.parametrize("log_format,display_attrs", [
     ("default", None),
-    ("default", "*"),
+    pytest.param("default", "*", marks=pytest.mark.xfail(reason="DS6886")),
     ("default", "nsmultiplexorcredentials"),
     ("json", None),
-    ("json", "*"),
+    pytest.param("json", "*", marks=pytest.mark.xfail(reason="DS6886")),
     ("json", "nsmultiplexorcredentials")
 ])
 def test_password_masking_multiplexor_credentials(topo, log_format, display_attrs):
@@ -368,10 +368,10 @@ def test_password_masking_multiplexor_credentials(topo, log_format, display_attr
 
 @pytest.mark.parametrize("log_format,display_attrs", [
     ("default", None),
-    ("default", "*"),
+    pytest.param("default", "*", marks=pytest.mark.xfail(reason="DS6886")),
     ("default", "nsDS5ReplicaCredentials"),
     ("json", None),
-    ("json", "*"),
+    pytest.param("json", "*", marks=pytest.mark.xfail(reason="DS6886")),
     ("json", "nsDS5ReplicaCredentials")
 ])
 def test_password_masking_replica_credentials(topo, log_format, display_attrs):
@@ -432,10 +432,10 @@ def test_password_masking_replica_credentials(topo, log_format, display_attrs):
 
 @pytest.mark.parametrize("log_format,display_attrs", [
     ("default", None),
-    ("default", "*"),
+    pytest.param("default", "*", marks=pytest.mark.xfail(reason="DS6886")),
     ("default", "nsDS5ReplicaBootstrapCredentials"),
     ("json", None),
-    ("json", "*"),
+    pytest.param("json", "*", marks=pytest.mark.xfail(reason="DS6886")),
     ("json", "nsDS5ReplicaBootstrapCredentials")
 ])
 def test_password_masking_bootstrap_credentials(topo, log_format, display_attrs):
diff --git a/dirsrvtests/tests/suites/plugins/entryusn_overflow_test.py b/dirsrvtests/tests/suites/plugins/entryusn_overflow_test.py
index a23d734ca..8c3a537ab 100644
--- a/dirsrvtests/tests/suites/plugins/entryusn_overflow_test.py
+++ b/dirsrvtests/tests/suites/plugins/entryusn_overflow_test.py
@@ -81,6 +81,7 @@ def setup_usn_test(topology_st, request):
     return created_users
 
 
+@pytest.mark.xfail(reason="DS6250")
 def test_entryusn_overflow_on_add_existing_entries(topology_st, setup_usn_test):
     """Test that reproduces entryUSN overflow when adding existing entries
 
@@ -232,6 +233,7 @@ def test_entryusn_overflow_on_add_existing_entries(topology_st, setup_usn_test):
     log.info("EntryUSN overflow test completed successfully")
 
 
+@pytest.mark.xfail(reason="DS6250")
 def test_entryusn_consistency_after_failed_adds(topology_st, setup_usn_test):
     """Test that entryUSN remains consistent after failed add operations
 
-- 
2.49.0

